### PR TITLE
cleanup remove old matcher code and update doku

### PR DESCRIPTION
**Pull request by heikofriedrich**
- remove old matcher node and doku about matching
- fixes #660

### CODE REVIEW

**Comment by fkleedorfer** [comment]
@heikofriedrich please leave a message here when the PR is ready for review

**Comment by heikofriedrich** [comment]
@fkleedorfer can be merged now

File: webofneeds/won-matcher/src/main/java/won/matcher/component/MatcherNodeURISource.java [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
- */
-
-package won.matcher.component;
-
-import java.util.Iterator;
-
-/**
- * Interface that is used to obtain a WON_BA node URI.
- */
-public interface MatcherNodeURISource
```
Comment (fkleedorfer) [inline webofneeds/won-matcher/src/main/java/won/matcher/component/MatcherNodeURISource.java]: We cannot delete this interface and its implementations: they are still used. I am sceptical if we should really delete won-matcher at all.. The bots will not work without them.

**Comment by heikofriedrich** [comment]
ok, i didnt see this when i did the build, shall we keep the won-matcher project and anyway delete won-matcher-solr and won-matcher-webapp?

**Comment by heikofriedrich** [comment]
@fkleedorfer didnt delte the won-matcher project => can be treviewed

**Comment by fkleedorfer** [comment]
I see - but now we deleted won-matcher and then added the code again as new, which means that we lose the whole edit history of that code. Not ideal.

**Comment by heikofriedrich** [comment]
ah ok, i will fix that

**Comment by heikofriedrich** [comment]
@fkleedorfer fixed

**Pull request merged**